host/src/FunctionsNetHost/Native/NativeExports.cs

```diff
@@ -10,24 +10,10 @@ namespace FunctionsNetHost
     public static class NativeExports
     {
         [UnmanagedCallersOnly(EntryPoint = "get_application_properties")]
-        public static int GetApplicationProperties(NativeHostData nativeHostData)
+        public static unsafe int GetApplicationProperties(NativeHostData* nativeHostData)
```

Comment:
@elinor-fung Just to confirm, Is this change breaking the contract with the API in the FunctionsNetHost? 

The code in https://github.com/Azure/azure-functions-dotnet-worker/tree/main/host (FunctionsNetHost - native placeholder) and code in https://github.com/Azure/azure-functions-dotnet-worker/tree/main/src (Source of [https://www.nuget.org/packages/Microsoft.Azure.Functions.Worker](https://www.nuget.org/packages/Microsoft.Azure.Functions.Worker) nuget package - which customer will add as a reference to their function app) are shipped independently. So we need to make sure that we are not breaking any contracts.

Reply:
This does not break the contract. The managed p/invoke used `out NativeHost` as the parameter:
https://github.com/Azure/azure-functions-dotnet-worker/blob/bf56b378eecc0499aa86a89ae33ae6b1692b9f2d/src/DotNetWorker.Grpc/NativeHostIntegration/NativeMethods.cs#L44-L45

On the unmanaged side, this should map to a pointer to the struct (`NativeHostData*` instead of `NativeHostData`). So this change actually makes the unmanaged side conform to what was expected by p/invoke.